src/broker/radix_tree.rs

```diff
@@ -0,0 +1,207 @@
+const EMPTY: usize = usize::MAX;
+
+#[derive(Debug)]
+pub struct RadixTree<T> {
+    nodes: Vec<NodeData<T>>,
+    root: usize,
+}
+
+#[derive(Debug)]
+pub struct NodeData<T> {
+    prefix: String,
+    first_child: usize,
+    next_sibling: usize,
+    values: Vec<T>,
+}
+
+impl<T: Clone> Default for RadixTree<T> {
+    fn default() -> Self {
+        RadixTree::new()
+    }
+}
+
+impl<T: Clone> RadixTree<T> {
+    pub fn new() -> Self {
```

Comment:
More a discussion on the direction we want to take styling for this project rather than a mandate to change code, but for types that have empty ctors and Default impls, do we want to simply omit the ctor and expect callers to use Default?

Reply:
i lean towards retaining empty `new()`'s even w/Default impls as other types may need custom constructor params so having `new()` everywhere provides a consistent interface

Reply:
Tabling for now, makes sense